src/book/mod.rs

```diff
@@ -332,25 +332,26 @@ fn determine_renderers(config: &Config) -> Vec<Box<Renderer>> {
 }
 
 /// Look at the `MDBook` and try to figure out what preprocessors to run.
-fn determine_preprocessors(config: &Config) -> Vec<Box<Preprocessor>> {
+fn determine_preprocessors(config: &Config) -> Result<Vec<Box<Preprocessor>>> {
+
+    let preprocess_list = match config.build.preprocess {
+        Some(ref p) => p,
+        // If no preprocessor field is set, default to the LinkPreprocessor. This allows you
+        // to disable the LinkPreprocessor by setting "preprocess" to an empty list.
+        None => return Ok(vec![Box::new(LinkPreprocessor::new())])
+    };
+
     let mut preprocessors: Vec<Box<Preprocessor>> = Vec::new();
 
-    if let Some(preprocess_array) = config.get("preprocess").and_then(|o| o.as_array()) {
-        for key in preprocess_array.iter() {
-            match key.as_str() {
-                Some(key) if key == "links" => {
-                    preprocessors.push(Box::new(LinkPreprocessor::new()))
-                }
-                _ => {}
-            }
+    for key in preprocess_list {
+        if key == "links" {
```

Comment:
What are your thoughts on turning this `if key == "links" {...} else { bail!() }` into a match statement? Now that preprocessors are easier to work with, I imagine we'll be adding a bunch more in the future.